rich/logging.py

```diff
@@ -131,7 +131,11 @@ def get_level_text(self, record: LogRecord) -> Text:
 
     def emit(self, record: LogRecord) -> None:
         """Invoked by logging."""
-        message = self.format(record)
+        try:
```

Comment:
Since there's a fair bit of logic below which may also fail, I'd lean towards wrapping the entire method implementation within the `try` block, similar to the approach used in `StreamHandler`:

```python
    def emit(self, record):
        """
        Emit a record.

        If a formatter is specified, it is used to format the record.
        The record is then written to the stream with a trailing newline.  If
        exception information is present, it is formatted using
        traceback.print_exception and appended to the stream.  If the stream
        has an 'encoding' attribute, it is used to determine how to do the
        output to the stream.
        """
        try:
            msg = self.format(record)
            stream = self.stream
            # issue 35046: merged two stream.writes into one.
            stream.write(msg + self.terminator)
            self.flush()
        except RecursionError:  # See issue 36272
            raise
        except Exception:
            self.handleError(record)
```

I've just tested your solution with `log.warning("hello {}", 123)` and found that another exception is raised from the following line:

```python
        message_renderable = self.render_message(record, message)
``` 

To keep the original implementation clean, you may want to consider something like this:

```python
    ...
    def _emit(self, record: LogRecord) -> None:
        ...
        # original emit implementation
        ...

    def emit(self, record: LogRecord) -> None:
        try:
            return self._emit(record)
        except Exception:
            self.handleError(record)
```

Reply:
Thanks for the feedback, I am gonna apply the changes